spec/lib/whitehall_importer_spec.rb

```diff
@@ -27,26 +28,74 @@
     end
   end
 
+  describe ".queue_migration_jobs" do
+    before do
+      @whitehall_migration = create(:whitehall_migration)
+      create_list(:whitehall_migration_document_import, 5, whitehall_migration_id: @whitehall_migration["id"], state: "pending")
+    end
+
+    it "queues a job for each listed document" do
+      WhitehallImporter.queue_migration_jobs(@whitehall_migration)
+      expect(WhitehallDocumentImportJob).to have_been_enqueued.exactly(5).times
+    end
+  end
+
+  describe ".mark_migration_completed" do
+    context "with only jobs that have finished running" do
+      before do
+        @whitehall_migration = create(:whitehall_migration)
+        @whitehall_migration_document_import_completed = create(:whitehall_migration_document_import, whitehall_migration_id: @whitehall_migration["id"], state: "completed")
+        @whitehall_migration_document_import_aborted = create(:whitehall_migration_document_import, whitehall_migration_id: @whitehall_migration["id"], state: "import_aborted")
+        @whitehall_migration_document_import_failed = create(:whitehall_migration_document_import, whitehall_migration_id: @whitehall_migration["id"], state: "import_failed")
+        @whitehall_migration_document_import_sync_failed = create(:whitehall_migration_document_import, whitehall_migration_id: @whitehall_migration["id"], state: "sync_failed")
```

Comment:
Convention would dictate using let arguments rather than creating these as ivars, there's various other ones of these scattered around this file.

Reply:
We've removed the assignments for the objects that we're not actually using and added a `let!` for those that need an assignment. This tidies it up a bit :)